src/register.jl

```diff
@@ -109,6 +109,172 @@ struct RegBranch
     error::Union{Nothing, String}
 end
 
+function update_package_data(pkg::Pkg.Types.Project, package_repo, package_path, tree_hash)
+    # package file
+    @debug("update package data: package file")
+    package_info = filter(((k,v),)->!(v isa Dict), Pkg.Types.destructure(pkg))
+    delete!(package_info, "version")
+    package_info["repo"] = package_repo
+    package_file = joinpath(package_path, "Package.toml")
+    write_toml(package_file, package_info)
+
+    # versions file
+    @debug("update package data: versions file")
+    versions_file = joinpath(package_path, "Versions.toml")
+    versions_data = isfile(versions_file) ? TOML.parsefile(versions_file) : Dict()
+
+    try
+        versions = sort!([VersionNumber(v) for v in keys(versions_data)])
+        #Base.check_new_version(versions, pkg.version)
+    catch ex
+        if isa(ex, ErrorException)
+            return RegBranch(pkg.name, pkg.version, branch, ex.msg)
+        else
+            rethrow(ex)
+        end
+    end
+
+    version_info = Dict{String,Any}("git-tree-sha1" => string(tree_hash))
+    versions_data[string(pkg.version)] = version_info
+
+    vnlist = sort([(VersionNumber(k), v) for (k, v) in versions_data])
+    vslist = [(string(k), v) for (k, v) in vnlist]
+
+    open(versions_file, "w") do io
+        TOML.print(io, OrderedDict(vslist))
+    end
+
+    # deps file
+    @debug("update package data: deps file")
+    deps_file = joinpath(package_path, "Deps.toml")
+    if isfile(deps_file)
+        deps_data = Pkg.Compress.load(deps_file)
+    else
+        deps_data = Dict()
+    end
+
+    @debug("Verifying package name and uuid in deps")
+    for (k, v) in pkg.deps
+        u = string(v)
+        if haskey(registry_data["packages"], u)
+            name_in_reg = registry_data["packages"][u]["name"]
+            if name_in_reg != k
+                err = "Error in `[deps]`: UUID $u refers to package '$name_in_reg' in registry but deps file has '$k'"
+                throw(err)
+            end
+        elseif haskey(BUILTIN_PKGS, k)
+            if BUILTIN_PKGS[k] != u
+                err = "Error in `[deps]`: UUID $u for package $k should be $(BUILTIN_PKGS[k])"
+                throw(err)
+            end
+        else
+            err = "Error in `[deps]`: Package '$k' with UUID: $u not found in registry or stdlib"
```

Comment:
It looks like this only works if all non-stdlib dependencies are within the same registry. It would be good to support other scenarios, in particular the case of a private registry that allows external dependencies both in stdlib and the General registry.

Reply:
This is fixed in #67